articles/azure-functions/durable/durable-functions-entities.md

```diff
@@ -19,28 +19,52 @@ Entity functions define operations for reading and updating small pieces of stat
+
+**Entity Communication**. Entities communicate with other entities, orchestrations and clients using messages that are implicitly sent via reliable queues. We use the following terminology to distinguish the difference between one-way and two-way communication: 
+
+* *Calling* an entity means we use two-way (round-trip) communication: we send an operation message to the entity, and then wait for the response message before continuing. The response message can provide a result value or an error result (such as a JavaScript error or a .NET exception), which is observed by the caller.
+* *Signaling* an entity means we use one-way (fire and forget) communication: we sends an operation message, but do not wait for a response. While the message is guaranteed to be eventually delivered, the sender does not know when, and cannot observe any result value or errors.
+
+Entities can be called only from within orchestrations.
+
+<a name="function-based-entity"></a>
+## Defining Entities
 
-Durable entities support two different programming models. The first model is a dynamic "functional" model where the entity is defined by a single function. The second model is an object-oriented model where the entity is defined by a class and methods. These models and the programming models for interacting with entities are described in the next sections.
+We currently offer two distinct APIs for defining entities.
 
-### Defining entities
+A **function-based syntax** where entities are represented as functions, and operations are explicitly dispatched by the application. This works well for entities with simple state, few operations, or a dynamic set of operations (such as in application frameworks). However, it can be tedious to maintain, as it does not catch type errors at compile time.
 
-There are two optional programming models for authoring durable entities. The following code is an example of a simple *Counter* entity implemented as a standard function. This function defines three *operations*, `add`, `reset`, and `get`, each of which operate on an integer state value, `currentValue`.
+A **class-based syntax** where entities and operations are represented by classes and methods. This produces more easily readable code and allows operations to be invoked in a type-safe way. The class-based syntax is just a thin layer on top of the function-based syntax, so both variants can be used interchangeably in the same application.
+
+### Function-Based Syntax
+
+The following code is an example of a simple *Counter* entity implemented as a durable function. This function defines three operations, `add`, `reset`, and `get`, each of which operate on an integer state value, `currentValue`.
 
 ```csharp
 [FunctionName("Counter")]
 public static void Counter([EntityTrigger] IDurableEntityContext ctx)
 {
-    int currentValue = ctx.GetState<int>();
+    int currentValue = ctx.GetState<int>(); // get or create state
```

Comment:
Is this behavior of `GetState<T>()` relevant to the developer? I can't think of what someone might do differently if they knew that `GetState<T>()` creates some default state.